### PR TITLE
Fast forward readme

### LINKED ISSUE #280: Fast Forward Documentation
It appears the README has fallen a bit out of date due to our insistence on adding code ;D. Let's make sure we're back up to par.

### PR DESCRIPTION
**Pull request by bnookala**
Documents the undocumented, fixes #280

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,18 +8,16 @@ the `vs-kubernetes` extension by @brendandburns and the `vs-helm` extension by @
...
-This extension assumes that you have a `Dockerfile` in the root directory of
-your project.
+This extension assumes you have the following binaries on your `PATH`:
 
-It also assumes that you have the following binaries on your `PATH`:
-
-   * `kubectl`
    * `docker`
    * `git`
+   * `kubectl` (optional)
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: We do offer to install it if we can't find it, so I'd say yes.

File: README.md [hunk 2]
```diff
@@ -66,26 +65,32 @@ your user or workspace settings.
...
+### Kubernetes
 
-### General commands
+#### General commands
 
    * `Kubernetes: Load` - Load a resource from the Kubernetes API and create a new editor window.
    * `Kubernetes: Get` - Get the status for a specific resource.
    * `Kubernetes: Follow Logs` - Follow logs for a pod in an output window.
    * `Kubernetes: Show Logs` - Get logs for a pod in an output window.
+   * `Kubernetes: Show Events` - Show events on a selected namespace.
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -66,26 +65,32 @@ your user or workspace settings.
...
+   * `Kubernetes: Follow Events` - Follow events on a selected namespace.
 
-### Commands while viewing a Kubernetes file
+#### Commands while viewing a Kubernetes manifest file
 
    * `Kubernetes: Explain` - Use the `kubectl explain ...` tool to annotate Kubernetes API objects
    * `Kubernetes: Create` - Create an object using the current document
    * `Kubernetes: Delete` - Delete an object contained in the current document.
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: we should add a section for commands that work for the resource explorer

File: README.md [hunk 4]
```diff
@@ -95,16 +100,32 @@ If you are building and running the extension from source, see [CONTRIBUTING.md]
    * `Kubernetes: Remove Debug` - Remove the deployment and/or service created for a `Kubernetes Debug (Launch)` session
    * `Kubernetes: Sync Working Copy to Cluster` - Checks out the version of the code that matches what is deployed in the cluster.  (This relies on Docker image versions also being Git commit IDs, which the extension does if you use the Run command to build the container, but which typically doesn't work for containers/deployments done by other means.)
 
-### Configuration commands
+#### Configuration commands
+
+   * `Kubernetes: Create Cluster` - Initiate the flow for creating an Azure Kubernetes Service cluster through the configured Azure CLI.
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -95,16 +100,32 @@ If you are building and running the extension from source, see [CONTRIBUTING.md]
...
+   * `Kubernetes: Add Existing Cluster` - Install and configure the Kubernetes command line tool (kubectl) from a cloud cluster, such as an Azure Container Service (ACS) or Azure Kubernetes Service (AKS) cluster
+   * `Kubernetes: Set as Current Cluster` - Select from a list of configured clusters to set the "current" cluster. Used for searching, displaying, and deploying Kubernetes resources.
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md [hunk 6]
```diff
@@ -95,16 +100,32 @@ If you are building and running the extension from source, see [CONTRIBUTING.md]
...
+   * `Kubernetes: Show Cluster Info` - For a cluster, show the status of Kubernetes Components (API Server, etcd, KubeDNS, etc.) in a terminal window.
+   * `Kubernetes: Use Namespace` - Select from a list of namespaces to set the "current" namespace. Used for searching, displaying, and deploying Kubernetes resources.
+
+#### ConfigMap & Secret commands
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md [hunk 7]
```diff
@@ -95,16 +100,32 @@ If you are building and running the extension from source, see [CONTRIBUTING.md]
...
+
+   * `Kubernetes: Add File` - Adds a file as a ConfigMap or a Secret
+   * `Kubernetes: Delete File` - Deletes a file from a ConfigMap or a Secret
+
+#### Miscellaneous commands
 
-   * `Kubernetes Add Existing Cluster` - Install and configure the Kubernetes command line tool (kubectl) from a cloud cluster, such as an Azure Container Service (ACS) or Azure Kubernetes Service (AKS) cluster
+   * `Kubernetes: Open Dashboard` - Opens the Kubernetes Dashboard in your browser.
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: who could forget about lovely ol' port forward?

File: README.md [hunk 8]
```diff
@@ -95,16 +100,32 @@ If you are building and running the extension from source, see [CONTRIBUTING.md]
...
 
+### Minikube
 
-### Miscellaneous commands
+[Minikube](https://github.com/kubernetes/minikube) runs a local, single node Kubernetes cluster inside a VM. Support is currently experimental, and requires
+Minikube tools to be installed and available on your PATH.
 
-   * `Kubernetes Open Dashboard` - Opens the Kubernetes Dashboard in your browser.
+   * `Kubernetes: Start minikube` - Starts Minikube
+   * `Kubernetes: Stop minikube` - Stopos Minikube
```
[the remark this answers is not in the document]
Comment (bnookala) [reply README.md]: fixed

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (itowlson) [reply README.md]: You mean apart from the fellow who wrote it?  *grin*

**Pull request merged**